scour/scour.py

```diff
@@ -3059,9 +3066,27 @@ def scourString(in_string, options=None):
       else:
          print("WARNING: {}".format(errmsg), file = options.ensure_value("stdout", sys.stdout))
 
-   # remove <metadata> if the user wants to
+   # remove descriptive elements
+   descriptiveElements = set()
```

Comment:
This is unnecessarily complex...
Just pass `options` to the function `removeDescriptiveElements(doc, options)` and do the required checking there.

You can then skip all the safety checks around line 914.

Also if you check for `remove_descriptive_elements` first you can potentially skip the check for the others and also do not require `Set.update()` (which consequently allows you to use a simple list).